scheduler/task.go

```diff
@@ -301,25 +301,28 @@ func (t *task) spin() {
 				} else {
 					consecutiveFailures = 0
 				}
-				if consecutiveFailures >= t.stopOnFailure {
-					taskLogger.WithFields(log.Fields{
-						"_block":               "spin",
-						"task-id":              t.id,
-						"task-name":            t.name,
-						"consecutive failures": consecutiveFailures,
-						"error":                t.lastFailureMessage,
-					}).Error(ErrTaskDisabledOnFailures)
-					// You must lock on state change for tasks
-					t.Lock()
-					t.state = core.TaskDisabled
-					t.Unlock()
-					// Send task disabled event
-					event := new(scheduler_event.TaskDisabledEvent)
-					event.TaskID = t.id
-					event.Why = fmt.Sprintf("Task disabled with error: %s", t.lastFailureMessage)
-					defer t.eventEmitter.Emit(event)
-					return
+				if t.stopOnFailure >= 0 {
```

Comment:
One last nit, and then I think this is ready to land.  Now that we don't have the extra state, would you mind combining these nested `if`s with an `&&`?